enhancements/machine-api/aws-imds-v2-support.md

```diff
@@ -0,0 +1,264 @@
+- As an OpenShift cluster administrator, I would like to change IMDS interaction settings for newly created machines in my cluster.
+
+### API Extensions
+
+#### AWSMachineProviderConfig changes
+
+MachineAPI design mostly replicates parameters which [AWS API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html) consumes.
+
+```go
+type AWSMachineProviderConfig struct {
+    // Existing fields will not be modified
+    ...
+    // MetadataServiceOptions allows users to configure instance metadata service interaction options.
+    // If nothing specified, default AWS IMDS settings will be applied.
+    // https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html
+    // +optional
+    MetadataServiceOptions *MetadataServiceOptions `json:"metadataServiceOptions,omitempty"`
+}
+```
+
+```go
+
+type InstanceMetadataHTTPTokens string
+
+const (
+    // InstanceMetadataHTTPTokensRequired enforces sending of a signed token header with any instance metadata retrieval (GET) requests.
+    // Enforces IMDSv2 usage.
+    InstanceMetadataHTTPTokensRequired = "required"
+    // InstanceMetadataHTTPTokensOptional allows IMDSv1 usage along with IMDSv2
+    InstanceMetadataHTTPTokensOptional = "optional"
+)
+
+// MetadataServiceOptions defines the options available to a user when configuring
+// Instance Metadata Service (IMDS) Options.
+type MetadataServiceOptions struct {
+    // HTTPTokens determines whether or not the host requires the use of authentication when interacting with the metadata service.
+    // When using authentication, this enforces v2 interaction method (IMDSv2) with the metadata service.
+    // Default: optional
+    // +kubebuilder:validation:Enum=required;optional
+    // +optional
+    HTTPTokens InstanceMetadataHTTPTokens `json:"httpTokens,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Install Config changes
+
+For supporting IMDSv2 during installation we could extend install-config with aws specific `metadataServiceOptions`
+section in the installer configuration.
+
+Example (origin is the [aws installation doc](https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-customizations.html#installation-aws-config-yaml_installing-aws-customizations)):
+
+```yaml
+apiVersion: v1
+baseDomain: example.com 
+controlPlane:   
+  hyperthreading: Enabled 
+  name: master
+  platform:
+    aws:
+      zones:
+      - us-west-2a
+      - us-west-2b
+      rootVolume:
+        iops: 4000
+        size: 500
+        type: io1
+      metadataService: # proposed section
+        httpTokens: required
+      type: m5.xlarge
+  replicas: 3
+compute: 
+- hyperthreading: Enabled 
+  name: worker
+  platform:
+    aws:
+      rootVolume:
+        iops: 2000
+        size: 500
+        type: io1
+      metadataService: # proposed section
+        httpTokens: required
```

Comment:
Just as a note, the capitalisation is not consistent with the rest of the install config API, this should really be `Required` as per conventions

Reply:
I would prefer to keep it consistent with MAPI values. Also, there is some other enums for various platform-specific which does not follow such convention.